phylo/src/optimisers/dna_model_optimiser.rs

```diff
@@ -58,48 +60,47 @@ impl<'a> DNAModelOptimiser<'a> {
 
     pub fn optimise_parameters(
         &self,
-        start_values: &DNASubstParams,
         optimise_freqs: FrequencyOptimisation,
     ) -> Result<(u32, DNASubstParams, f64)> {
+        let start_values = self.likelihood_cost.model.params.clone();
         let model_type = start_values.model_type;
         info!("Optimising {} parameters.", model_type);
-        let param_sets = DNASubstParams::parameter_definition(model_type);
+        let param_sets = DNASubstParams::parameter_definition(&model_type);
         let start_values = match model_type {
             JC69 | K80 => start_values,
             _ => {
                 match optimise_freqs {
                     Fixed => start_values,
-                    Empirical => &self.set_empirical_frequencies(start_values),
+                    Empirical => self.set_empirical_frequencies(&start_values),
                     Estimated => {
                         warn!("Stationary frequency estimation not available, falling back on empirical.");
-                        &self.set_empirical_frequencies(start_values)
+                        self.set_empirical_frequencies(&start_values)
                     }
                 }
             }
         };
-        self.run_parameter_brent(start_values, param_sets)
+        self.run_parameter_brent(&start_values, param_sets)
     }
 
     fn run_parameter_brent(
         &self,
         start_values: &DNASubstParams,
-        param_sets: Vec<(&str, Vec<Parameter>)>,
+        param_sets: Vec<(&str, Vec<DNAParameter>)>,
     ) -> Result<(u32, DNASubstParams, f64)> {
         let mut prev_logl = f64::NEG_INFINITY;
-        let mut opt_logl = self
-            .likelihood_cost
-            .compute_log_likelihood(&make_dna_model(start_values.clone()))
-            .0;
+        let mut opt_logl = self.likelihood_cost.compute_log_likelihood().0;
         info!("Initial logl: {}.", opt_logl);
         let mut opt_params = start_values.clone();
+        let mut model = DNASubstModel::create(&opt_params);
```

Comment:
thinking out loud: `::create` in Rust conventionally is used to denote that there is an external resource created. Is it the case here? or should `::create` be renamed to `::new` (or `::new_with_params`)?

Reply:
I will rethink these names. Potentially with a better API only one of new/create will be necessary (currently there's both and it's pointless because they do kind of the same thing).